sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -753,6 +753,9 @@ object HiveTypeCoercion {
         case (DateType, TimestampType) => Cast(e, TimestampType)
         case (TimestampType, DateType) => Cast(e, DateType)
 
+        // Implicit cast for functions like utc_timestamp
+        case (_: NumericType, TimestampType) => Cast(e, TimestampType)
```

Comment:
Currently, we cast numeric type to timestamp type in different ways, for example, byte/short/int/long will be milliseconds, decimal/float/double will be seconds.

Also, we use seconds when we turn timestamp into long.

When utc_timestamp() work with long, is it used as seconds or milliseconds ?